src/Iterators-Tests/GeneratorWrapperTest.class.st

```diff
@@ -0,0 +1,33 @@
+"
+A GeneratorWrapperTest is a test class for testing the behavior of GeneratorWrapper
+"
+Class {
+	#name : #GeneratorWrapperTest,
+	#superclass : #TestCase,
+	#category : #'Iterators-Tests-Wrappers'
+}
+
+{ #category : #tests }
+GeneratorWrapperTest >> testEmptyGenerator [
+	| res |
+	res := (GeneratorWrapper generator: [:gen | ]) > Array.
+	self assert: res isEmpty 
+]
+
+{ #category : #tests }
+GeneratorWrapperTest >> testFiveEvenNumbersGenerator [
+	| res |
+	res := (GeneratorWrapper generator: [:gen | 1 to: 5 do:[:i | gen yield: i*2]]) > Array.
+	self assert: res equals: #(2 4 6 8 10)
+]
+
+{ #category : #tests }
+GeneratorWrapperTest >> testPipingGenerator [
+	| res |
+	res := ((1 to: 3) iterator
+		|
```

Comment:
I would like that the `|` is not used in the kernel of Iterators, this is part of the DSL that can be changed.
I recommand to use `#decoratedBy:` instead.